### PR TITLE
discourseAllPlugins: init discourse-checklist, discourse-data-explorer, discourse-migratepassword

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change

discourse-checklist: allows rendering interactive checkboxes for TODO style posts

image

discourse-data-explorer: allows admins to run queries, useful for making it easier to export surveys

image

discourse-migratepassword: useful when migrating from another forum type while not requiring users to remake their password. For example, phpbb.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dpausp — CHANGES_REQUESTED** [review]
Great, I wanted the checklist plugin too. Didn't know about data-explorer before but it looks helpful!

Can confirm that discourse-checklist and discourse-data-explorer work as expected (tested with Discourse 2.8.0.beta4 on 21.05). I didn't check the functionality of discourse-migratepassword but it loads correctly and appears in the admin settings.

The new plugins should be added to [update.py](https://github.com/NixOS/nixpkgs/blob/9207d0909e6ff00912081c1571437f715d7ed41e/pkgs/servers/web-apps/discourse/update.py#L190). discourse-calendar is also missing there.

**Comment by ryantm** [comment]
@dpausp Thanks for the review! I updated the update.py script to include the missing plugins. I couldn't easily test the updater though because it makes use of the experimental nix CLI and some changes in that latest nix version have broken it.

File: pkgs/servers/web-apps/discourse/update.py [hunk 1]
```diff
@@ -187,9 +187,13 @@ def update_plugins():
 
     """
     plugins = [
+        {'name': 'discourse-calendar'},
         {'name': 'discourse-canned-replies'},
+        {'name': 'discourse-checklist'},
+        {'name': 'discourse-data-explorer'},
         {'name': 'discourse-github'},
         {'name': 'discourse-math'},
+        {'name': 'discourse-migratepassword'},
```
Comment (dpausp) [inline pkgs/servers/web-apps/discourse/update.py:196]: this is from a different org, should be `{'name': 'discourse-migratepassword', 'owner': 'discoursehosting'},`

**Review by dpausp — CHANGES_REQUESTED** [review]
update.py fails for discourse-migratepassword, needs owner setting.

File: pkgs/servers/web-apps/discourse/update.py (hunk 1 above)
Comment (ryantm) [reply pkgs/servers/web-apps/discourse/update.py]: Should be resolved in latest force push.

**Review by dpausp — APPROVED** [review]
LGTM, thanks. Nice to see that our plugin collection is growing :)

I'm working on some plugins: discourse-assign, discourse-chat-integration, discourse-docs, discourse-openid-connect, discourse-prometheus and discourse-saved-searches.

**Pull request merged**